src/Emogrifier/HtmlProcessor/HtmlPruner.php

```diff
@@ -41,4 +41,47 @@ public function removeInvisibleNodes()
 
         return $this;
     }
+
+    /**
+     * Removes classes that are no longer required (e.g. because there are no longer any CSS rules that reference them)
+     * from `class` attributes.
+     *
+     * Note that this does not inspect the CSS, but expects to be provided with a list of classes that are still in use.
+     *
+     * This method also has the (presumably beneficial) side-effect of minifying (removing superfluous whitespace from)
+     * `class` attributes.
+     *
+     * @param string[] $classesToKeep list of class names that should not be removed
+     *
+     * @return self fluent interface
+     */
+    public function removeRedundantClasses(array $classesToKeep)
```

Comment:
Would it make sense to have give `$classesToKepp` a default value of `[]`?

Reply:
Yes, I think so...